### DISCUSSION TITLE
How to change remote address dynamically from code?

### DISCUSSION THREAD

**Discussion by trandzik** [FantasticFiasco/serilog-sinks-udp#200] [Q&A] [answered]
Hello,
please may I ask whether it is possible to change remote address dynamically from code? I have configured UDP sink based on a simple example in this repo (and it works great), but I would like to change the address at a later point in time.
~~~C#
Serilog.ILogger log = new LoggerConfiguration()
  .MinimumLevel.Verbose()
  .WriteTo.Udp("localhost", 7071, AddressFamily.InterNetwork)
  .CreateLogger();
~~~

I am not using [Serilog.Settings.Configuration](https://github.com/serilog/serilog-settings-configuration) as my use-case doesn't work well with external files. Please what would be preferred approach for doing so? Thanks.

**Comment by FantasticFiasco** [comment, marked as answer]
Hi @trandzik!

Are you using the Singleton instance `Log.Logger`? If so, I think you can replace it with a new logger once you detect that you need to change the address.

**Reply by trandzik** [reply]
Hi @FantasticFiasco,

thank you very much for reply. Yes, I am using `Log.Logger`, this sounds like a good approach, I will try that.

I am wondering whether you think some cleanup is needed on such change? I guess I should call `Log.CloseAndFlush()` before doing the swap?

**Reply by FantasticFiasco** [reply]
That sounds reasonable, since [`UdpSink`](https://github.com/FantasticFiasco/serilog-sinks-udp/blob/db31138b22caa309453d9e109bc638afa6692f10/src/Serilog.Sinks.Udp/Sinks/Udp/Private/UdpSink.cs#L30) implements `IDisposable`, and `CloseAndFlush` is [disposing the sink](https://github.com/serilog/serilog/blob/73325f64a1e4192751c3c067a300c1f541a7d443/src/Serilog/Log.cs#L55). You are correct.

**Reply by trandzik** [reply]
Thanks @FantasticFiasco, it works nicely 👍
